spec/lib/artifact-search.spec.js

```diff
@@ -0,0 +1,125 @@
+            ]);
+            responseHandlerMock.resolveLink.andCallFake((options, val) => {
+                return val;
+            });
+
+            return instance.search().then((urls) => {
+                expect(urls).toEqual(jasmine.any(Array));
+            });
+        });
+        it("calls responseHandler.handleErrorResponse on error", () => {
+            requestPromiseMock.post.andReturn(bluebird.reject());
+
+            return instance.search().then(() => {
+                expect(responseHandlerMock.handleErrorResponse).toHaveBeenCalled();
+            });
+        });
+        describe(".validateSearchItem()", () => {
+            var searchParameters;
+
+            beforeEach(() => {
+                spyOn(bluebird, "reject").andCallThrough();
+                spyOn(bluebird, "resolve").andCallThrough();
+                responseHandlerMock.resolveLinkHeaders.andReturn([]);
+            });
+            it("rejects if the query violates the regex", () => {
+                searchParameters = {
+                    search: "\\=Invalid Query"
+                };
+
+                return instance.search(searchParameters).then(jasmine.fail, () => {
+                    expect(bluebird.reject).toHaveBeenCalled();
+                });
+            });
+            it("resolves with a valid query", () => {
+                searchParameters = {
+                    search: "Valid=Query"
+                };
+
+                return instance.search(searchParameters).then(() => {
+                    expect(bluebird.resolve).toHaveBeenCalled();
+                });
+            });
+            it("rejects if the query isn't a string", () => {
+                searchParameters = {
+                    search: {
+                        query: "This won't work"
+                    }
+                };
+
+                return instance.search(searchParameters).then(jasmine.fail, () => {
+                    expect(bluebird.reject).toHaveBeenCalled();
+                });
+            });
+        });
+        describe(".validateSearch()", () => {
+            var searchParameters;
+
+            beforeEach(() => {
+                responseHandlerMock.resolveLinkHeaders.andReturn([]);
+            });
+            it("resolves the promise if the query list is falsy", () => {
+                spyOn(bluebird, "resolve").andCallThrough();
+
+                return instance.search().then(() => {
+                    expect(bluebird.resolve).toHaveBeenCalled();
+                });
+            });
+            it("turns the query list into an array if it isn't already", () => {
+                spyOn(bluebird, "map").andReturn(bluebird.resolve());
+                searchParameters = {
+                    search: "Single Query"
+                };
+
+                return instance.search(searchParameters).then(() => {
+                    expect(bluebird.map).toHaveBeenCalled();
```

Comment:
It called bluebird.map, but did it. Maybe the name should be calls map to turn list into array or something, since you aren't explicitly testing the result is an array.